### PR TITLE
Implementing Yolo  v8+ Dependencies 

### PR DESCRIPTION
**Pull request by arjun-sachar**
# Description

This change allows for YOLO object detection models to be accepted by ART wrappers by handling Loss Dictionary when the expectation from torch is to receive a scalar. 

Fixes # (issue)

## Type of change

Please check all relevant options.

- [x] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [x] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [x] Was able to run docker Yolo file to validate changes. 
- [ ] Test B

**Test Configuration**:
- OS
- Python version
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes have been tested using both CPU and GPU devices

### CODE REVIEW

File: art/estimators/object_detection/pytorch_yolo.py [hunk 1]
```diff
@@ -16,16 +16,17 @@
...
-This module implements the task specific estimator for PyTorch YOLO v3 and v5 object detectors.
+This module implements the task specific estimator for PyTorch YOLO v3, v5, v8+ object detectors.
 
 | Paper link: https://arxiv.org/abs/1804.02767
 """
 from __future__ import annotations
 
 import logging
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Optional, Union
```
[the remark this answers is not in the document]
Comment (arjun-sachar) [reply art/estimators/object_detection/pytorch_yolo.py]: Done

File: art/estimators/object_detection/pytorch_yolo.py [hunk 2]
```diff
@@ -16,16 +16,17 @@
...
 
 import numpy as np
+import torch
```
[the remark this answers is not in the document]
Comment (arjun-sachar) [reply art/estimators/object_detection/pytorch_yolo.py]: Done

File: tests/estimators/object_detection/test_pytorch_yolo_loss_wrapper.py [hunk 3]
```diff
@@ -0,0 +1,324 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2022
+#
+# Test for PyTorchYoloLossWrapper
```
[the remark this answers is not in the document]
Comment (arjun-sachar) [reply tests/estimators/object_detection/test_pytorch_yolo_loss_wrapper.py]: Done

File: tests/estimators/object_detection/test_pytorch_yolo.py [hunk 4]
```diff
@@ -322,3 +322,39 @@ def test_patch(art_warning, get_pytorch_yolo):
 
     except ARTTestException as e:
         art_warning(e)
+
+
+def test_import_pytorch_yolo_loss_wrapper():
+    import torch
+    from art.estimators.object_detection.pytorch_yolo_loss_wrapper import PyTorchYoloLossWrapper
+
+    class DummyModel(torch.nn.Module):
```
[the remark this answers is not in the document]
Comment (arjun-sachar) [reply tests/estimators/object_detection/test_pytorch_yolo.py:331]: Done
